src/KOKKOS/sna_kokkos_impl.h

```diff
@@ -223,13 +234,14 @@ void SNAKokkos<DeviceType>::grow_rij(int newnatom, int newnmax)
   inside = t_sna_2i("sna:inside",natom,nmax);
   wj = t_sna_2d("sna:wj",natom,nmax);
   rcutij = t_sna_2d("sna:rcutij",natom,nmax);
+  element = t_sna_2i("sna:rcutij",natom,nmax);
   dedr = t_sna_3d("sna:dedr",natom,nmax,3);
 
-  blist = t_sna_2d_ll("sna:blist",idxb_max,natom);
+  blist = t_sna_2d_ll("sna:blist",idxb_max*ntriples,natom);
   //ulisttot = t_sna_2c("sna:ulisttot",natom,idxu_max);
-  ulisttot = t_sna_2c_ll("sna:ulisttot",idxu_max,natom);
+  ulisttot = t_sna_2c_ll("sna:ulisttot",idxu_max*nelements,natom);
```

Comment:
This may not give optimal memory access since the flattening loop assumes a memory layout which is the same for CPUs and GPUs. It may be better to make `elements` another dimension to the array so Kokkos can change the layout, but this could also hurt performance when `nelements=1` as in the ECP benchmarks.

Reply:
So I'm fine with leaving it the way it is.

Reply:
Agreed.

Reply:
We can try out adding element information to idxu in the future. We definitely will not need an extra dimension.

Reply:
OK I'm marking this point as resolved.

Reply:
From an instruction count standpoint: Since all the Views are dynamically sized as opposed to statically sized, I believe making `elements` another dimension vs the flattened index is a performance wash---it's another fused integer multiply add, whether or not it's explicit in the functors.

Whether or not `idxu` should be faster than `nelements` (and the appropriate generalizations for `zlist`, `blist`, `ylist`) _in general_ is an interesting question related to cache reuse. For `nelements`, etc, `==1` that's a wash.